### ISSUE TITLE
Paid Module LDAP Authentication can only import Person or inetOrgPerson

### ISSUE THREAD

**Issue by maifeld** [freescout-helpdesk/freescout#842]
Yesterday, I installed FreeScout 1.6.2 using Softaculous and bought some modules including LDAP Integration 1.0.8.

"Connect & Fetch Attributes" worked the first time because of your note "User name (e.g. admin) without Bind DN." so thanks.
"Attributes Mapping" is also well-done.  I've had to do a lot of messing with "DNs and Filters", though.
I have a problem that's most-like [Not using DN](https://github.com/freescout-helpdesk/freescout/issues/371)  #371 

It seems that LDAP will present results for, but FreeScout will not **import** anything _except_ inetOrgPerson objectclasses.  We have accounts I would like to import in "ou=Admins,dc=example,dc=com" which are customPerson objectclasses (including cn, gn, sn, and mail, et al.) with RDN=uid, extended from person.  I created some test accounts as children of ou=Admin that are person or inetOrgPerson accounts and inetOrgPerson will import if it has an email address.  I even tried to hack your apparent internal filter (objectClass=person) with:
"ou=Admins,dc=example,dc=com(objectclass=*)"

Then, the ldap.log shows the three customPerson accounts are found:
conn=4689 op=1 SRCH base="ou=Admins,dc=example,dc=com" scope=2 deref=0 filter="(&(objectClass=*)(objectClass=person))"
conn=4689 op=1 SRCH attr=*
conn=4689 op=1 SEARCH RESULT tag=101 err=0 nentries=3 text=

**But** FreeScout concurrently shows a notification window that these 3 entries are _ignored_:
0 user(s) will be imported.
0 user(s) will be updated.
0 user(s) import will fail.

I would like to use FreeScout's LDAP module to import objectclasses beyond just inetOrgPerson.

**Comment by freescout-helpdesk** [comment]
We will check what can be done.

**Comment by freescout-helpdesk** [comment]
Now in LDAP Module 1.0.9 if any filter is specified for DN, default filter is not used.

Import all objectclasses:
```
dc=example,dc=com(objectclass=*)
```

Import specific objectclass:
```
dc=example,dc=com(objectclass=customperson)
```

**Comment by maifeld** [comment]
Amazingly-fast response, thanks!  I should send you some more money!
I installed version 1.0.9 of the LDAP module.

I like the Default Filters pulldown you made.  That's also very helpful.
I see that it now gets custom attributes from my custom objectclasses during Fetch and can put them in First/Last under Attributes Mapping.  Cool!

I tried nickname (one of my custom attributes in customPerson) for the first_name and got a SQL error because some inetOrgPersons **didn't** have this attribute, you may want to construct a cleaner error message:
Unable to import user Firstname Lastname: SQLSTATE[23000]: Integrity constraint violation: 1048 Column 'first_name' cannot be null (SQL: insert into `users` (`email`, `first_name`, `last_name`, `password`, `invite_state`, `updated_at`, `created_at`) values (name@example.com, , Lastname, $2y$10$JeQRZBzD/Fcl8jCT1eg/1uQtjrm/gbmbdsCsf6INrCIjoPvBzi90m, 1, 2020-10-29 14:04:01, 2020-10-29 14:04:01))

I did more testing in this module with the Filters and DN.  LDAP appears to provide the right nentries in ldap.log (correct numbers of entries given the ou and objectclass I set), but I'm not seeing them available for Import/Update/Failed.
I tried 10 different combinations of ou and objectclass.  The ldap.log shows the appropriate nentries= as described in my OP.  LDAP Module's notification shows 0/0/0, except for whatever inetOrgPerson objects it catches.

Progress, but not there yet.  Should I DM you my .schema, or give you a login on my LDAP server?
Thanks for your continued support.

**Comment by freescout-helpdesk** [comment]
Did you manage to find a solution?

**Comment by maifeld** [comment]
I moved on to other LDAP integrations in the mean time.  I also saw some version updates and didn't want to bug you when you were busy.

I presume I'll have to discard my customPersons and just use inetOrgPersons instead?  Or, maybe SUP my customPerson to inetOrgPerson instead of person?  LDAP provides FreeScout the right number of entries (per the log), but then FreeScout doesn't show them in Import/Update/Failed.  I presume there's yet another filter in the LDAP module that removes them - everything but inetOrgPersons.

My organization likes FreeScout, but with our LDAP configuration and FreeScout's LDAP module talking past each other, we'll have to modify something to move forward.

**Comment by maifeld** [comment]
I tried inetOrgPerson as a SUPerior to my customPerson, but that didn't seem to matter.  Nor did the objectclass=* or =anything make any difference in filtering.  

The FreeScout LDAP module will only import an entry if the entry has inetOrgPerson included as a structural objectclass.  I tried with/without and that one change will make an entry Import/not.  I think it's customary to only have _one_ structural objectclass.  I'll have to add inetOrgPerson to all the customPerson entries to have them show up in FreeScout.
